homeassistant/components/asuswrt/router.py

```diff
@@ -22,10 +32,71 @@
     CONF_SSH_KEY,
     DEFAULT_DNSMASQ,
     DEFAULT_INTERFACE,
+    DOMAIN,
     PROTOCOL_TELNET,
 )
 
 CONF_REQ_RELOAD = [CONF_DNSMASQ, CONF_INTERFACE, CONF_REQUIRE_IP]
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AsusWrtDevInfo:
+    """Representation of a AsusWrt device info."""
+
+    def __init__(self, mac, name=None):
+        """Initialize a AsusWrt device info."""
+        self._mac = mac
+        self._name = name
+        self._ip_address = None
+        self._last_activity = None
+        self._connected = False
+
+    def update(self, dev_info=None, consider_home=0):
+        """Update AsusWrt device info."""
+        utc_point_in_time = dt_util.utcnow()
+        if dev_info:
+            if not self._name:
+                self._name = dev_info.name or self._mac.replace(":", "_")
+            self._ip_address = dev_info.ip
+            self._last_activity = utc_point_in_time
+            self._connected = True
+
+        elif self._connected:
+            self._connected = (
+                utc_point_in_time - self._last_activity
+            ).total_seconds() < consider_home
+            self._ip_address = None
+
+    @property
+    def is_connected(self):
+        """Return connected status."""
+        return self._connected
+
+    @property
+    def mac(self):
+        """Return device mac address."""
+        return self._mac
+
+    @property
+    def name(self):
+        """Return device name."""
+        return self._name
+
+    @property
+    def ip_address(self):
+        """Return device ip address."""
+        return self._ip_address
+
+    @property
+    def last_activity(self):
+        """Return device last activity."""
+        if not self._last_activity:
+            return None
+        full_date = self._last_activity
+        round_date = full_date - timedelta(microseconds=full_date.microsecond)
```

Comment:
Use `timespec` parameter in datetime.isoformat and set it to `"milliseconds"` to truncate the microseconds.

https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat